flink/v1.13/flink/src/main/java/org/apache/iceberg/flink/sink/IcebergStreamWriter.java

```diff
@@ -83,9 +83,18 @@ public void dispose() throws Exception {
   @Override
   public void endInput() throws IOException {
     // For bounded stream, it may don't enable the checkpoint mechanism so we'd better to emit the
-    // remaining
-    // completed files to downstream before closing the writer so that we won't miss any of them.
+    // remaining completed files to downstream before closing the writer so that we won't miss any
+    // of them.
+    // Note that if the task is not closed after calling endInput, checkpoint may be triggered again
+    // causing files to be sent repeatedly, so we need to set write to null after the file is sent.
+
+    if (writer == null) {
+      return;
+    }
```

Comment:
nit: We usually add empty lines after a block

Reply:
Has been revised